### DISCUSSION TITLE
MFXTextField addListener for changed not works properly

### DISCUSSION THREAD

**Discussion by Roccoxx** [palexdev/MaterialFX#167] [Archived]
Hello good afternoon, I have a problem when it comes to only allowing the entry of numbers, normally this code works in a TextField but for some reason I can't with an MFXTextField

```
SExtractionTextField.textProperty().addListener(new ChangeListener<String>() {
			@Override
			public void changed(ObservableValue<? extends String> getText, String oldValue, String newValue) {				
				if (!newValue.matches("\\d*(\\.\\d*)?")) {
					SExtractionTextField.setText(oldValue);
					System.out.println(getText + " " + oldValue + " " + newValue);
					System.out.println(SExtractionTextField.getText());
					//SExtractionTextField.clear();			
				}
			}
		});
```

**output**
`StringProperty [bean: MFXTextField[id=SExtractionTextField, styleClass=mfx-text-field], name: text, value: 3.75] 3.75 3.75a
3.75
StringProperty [bean: MFXTextField[id=SExtractionTextField, styleClass=mfx-text-field], name: text, value: 3.75] 3.75 3.75aa
3.75
`

**image**
![image](https://user-images.githubusercontent.com/53314419/160417985-7a66427c-c419-4c94-a89f-4f6f80021ff7.png)

apparently the text of the textfield is not updated, I have tried different ways but I cannot achieve my objective of preventing letters or more than a colon "."

Thanks for read

**Comment by palexdev** [comment]
If you need to limit the characters to only numbers than you should use a TextFormatter as suggested here: [StackOverflow](https://stackoverflow.com/questions/7555564/what-is-the-recommended-way-to-make-a-numeric-textfield-in-javafx)

Remember to use the delegate method though or it won't work
